test/Microsoft.AspNet.Tooling.Razor.Test/VisualStudioProtocolTest.cs

```diff
@@ -0,0 +1,334 @@
+                    }
+
+                    _prefix = value;
+                }
+            }
+            public string TagName
+            {
+                get
+                {
+                    return _tagName;
+                }
+                set
+                {
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(nameof(value));
+                    }
+
+                    _tagName = value;
+                }
+            }
+            public string FullTagName
+            {
+                get
+                {
+                    return Prefix + TagName;
+                }
+            }
+            public string TypeName
+            {
+                get
+                {
+                    return _typeName;
+                }
+                set
+                {
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(nameof(value));
+                    }
+
+                    _typeName = value;
+                }
+            }
+            public string AssemblyName
+            {
+                get
+                {
+                    return _assemblyName;
+                }
+                set
+                {
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(nameof(value));
+                    }
+
+                    _assemblyName = value;
+                }
+            }
+            public IEnumerable<VisualStudioTagHelperAttributeDescriptor> Attributes
+            {
+                get
+                {
+                    return _attributes;
+                }
+                set
+                {
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(nameof(value));
+                    }
+
+                    _attributes = value;
+                }
+            }
+            public IEnumerable<string> RequiredAttributes
+            {
+                get
+                {
+                    return _requiredAttributes;
+                }
+                set
+                {
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(nameof(value));
+                    }
+
+                    _requiredAttributes = value;
+                }
+            }
+            public IEnumerable<string> AllowedChildren { get; set; }
+            public string RequiredParent { get; set; }
+            public VisualStudioTagStructure TagStructure { get; set; }
+            public VisualStudioTagHelperDesignTimeDescriptor DesignTimeDescriptor { get; set; }
+        }
+        public enum VisualStudioTagStructure
+        {
+            Unspecified,
+            NormalOrSelfClosing,
+            WithoutEndTag
```

Comment:
Corner cases but the test only confirms the `int` value for `TagStructure.NormalOrSelfClosing` remains `1`. It won't catch new `TagStructure` values, changing `Unspecified` to equal `-1` or changing `WithoutEndTag` to equal `37`.